generators/app/files/CONTRIBUTING.md

```diff
@@ -2,9 +2,124 @@
 
 Please follow these guidelines for contributing to this repository.
 
+## Branches
+
+**Never commit directly into `master` or other long running branches.**
+Create small branches for *each* topic that you work on (individual features, bug fixes, and so on).
+When the branch is complete, [create a pull request](https://help.github.com/articles/using-pull-requests/)
+against `master`.
+
+Branches should be per topic, and thus *small* and *short-lived*. The goal is to be able to
+merge back into `master` frequently, so that the team can access your changes,
+and so that you stay in closer sync with the trunk.
+
+Remember to *frequently* [rebase](http://git-scm.com/book/ch3-6.html) your branch,
+to get the latest changes from `master`. This will give you access to any new
+features and utilities, plus make merging upstream (much) easier.
+
+### Naming
+
+Branch names should be:
+
+- Descriptive and concise (pithy, even)
+- All lowercase
+- Separated by *underscores*
+
+| Bad       | Good                |
+|-----------|---------------------|
+| `pnx`     | `init_phoenix`      |
+| `logo`    | `logo_v2`           |
+| `linter`  | `ci_linter_task`    |
+| `sockets` | `setup_web_sockets` |
+| `#42`     | `switch_to_adga`    |
+
+### Long-Running & Integration Branches
+
+In the edge case that a series of features are tightly coupled, and it is impractical to decouple them:
+
+1. Create a long-running "integration branch" from `master`
+2. Branch sub-features from the integration branch
+  - *Do not commit directly into the long-running branch directly*
+3. Frequently rebase sub-branches from the integration branch, and the feature branch from `master`
+4. When the sub-feature is complete, pull request against integration
+5. When the all coupled features are complete and merged into the integration branch,
+   pull request the integration branch against `master`
+
+It may be helpful to create a checklist in the integration branch's description
+of the subtasks that belong to it.
+
```

Comment:
Should we establish some best-practices around naming long-running branches?

It's common to see long running branches named `feature/master` with sub-feature branches named `feature/sub-feature`. This pattern gives nice context to the current hierarchy of what you're working on but won't work if a branch named `features` exists ([explination](https://coderwall.com/p/qkofma/a-caution-about-git-branch-names-with-s)).